Enabling iOS App Center tests

trwalke opened the pull request:
Enabling iOS App Center tests.
Refactoring current ui test project structure to reduce the number of concurrent agents need for App Center testing.

Commit:
Adding xamarin azure pipeline configuration

Commit:
Creating iOS specific solution

Commit:
Configuring iOS build

Commit:
Stripping projects from iOS solution

Commit:
updating configuration

Commit:
Configuring app

Commit:
Updating app configuration to target net standard

Commit:
temporary fix for ios testing

Commit:
Disabling MSAL tests

Commit:
adding iOS id

Commit:
Enabling calabash in iOS adal app

Commit:
Adding xamarin test cloud agent nuget to ios app

Commit:
Forcing calabash to always be on

Commit:
re enabling the ENABLE_TEST_CLOUD

Commit:
Adding variable to enable macOS build

Commit:
Reqolving configuration issue with source projects

Commit:
Refactoring conditional build targets

Commit:
Refactoring targets

Commit:
Removing macOS targets

Commit:
Fixing project tags

Commit:
Merge remote-tracking branch 'origin/dev' into trwalke/iOSAppCenterTest

Commit:
Adding target back into build

Commit:
Refactoring tests to consolidate appcenter test tasks.
adding new solution for MacOS build.

Commit:
Removing unused files.

henrik-me commented on adal/devApps/XFormsApp.iOS/AppDelegate.cs:
```diff
@@ -52,6 +52,9 @@ public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsAppli
         public override bool FinishedLaunching(UIApplication app, NSDictionary options)
         {
             global::Xamarin.Forms.Forms.Init();
+            #if ENABLE_TEST_CLOUD
```
>ENABLE_TEST_CLOUD [](start = 16, length = 17)

when is this set? #Closed

trwalke replied:
This is set externally by the build definition #Closed

henrik-me commented on adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidMSALTests.cs:
```diff
@@ -81,26 +81,9 @@ public void AcquireTokenTest()
-            _msalMobileTestHelper.CoreMobileTestHelper.VerifyResult(xamarinController);
+            var labResponse = LabUserHelper.GetLabResponseWithDefaultUser();
 
+            _msalMobileTestHelper.PromptBehaviorTestHelper(xamarinController, labResponse);
```
>_msalMobileTestHelper.PromptBehaviorTestHelper(xamarinController, labResponse); [](start = 12, length = 79)

Does this set consent by default? That is not obvious to me. #Closed

trwalke replied:
yes. however, it should allow you to set whatever behavior you want. ill update. #Closed

bgavrilMS commented on adal/devApps/XFormsApp/App.cs:
```diff
@@ -52,7 +52,7 @@ protected override void OnStart()
             AppCenter.Start("android=3ea693de-dcb4-4b65-a1b5-04ad2f71ada0;" +
                   "uwp={Your UWP App secret here};" +
-                  "ios={Your iOS App secret here}",
+                  "ios={4be201cc-2027-4045-b535-d57de5ff36c7}",
```
Should this be checked in? #Closed

trwalke replied:
This is simply the appID of our application in appcenter and it is used for identification purposes only. All appcenter credentials are stored securely elsewhere. #Pending

henrik-me replied:
nit: should the app id be in KeyVault? Allowing it to be changed outside of the test environment?

---
In reply to: [235211318](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1418#discussion_r235211318) [](ancestors = 235211318)

henrik-me commented on core/src/Platforms/iOS/WebviewBase.cs:
```diff
@@ -28,7 +28,7 @@
 using SafariServices;
 using Foundation;
-using AuthenticationServices;
+//using AuthenticationServices;
```
why keeping the commented out stuff?

henrik-me commented on core/src/Platforms/iOS/SystemWebview/SystemWebUI.cs:
```diff
@@ -66,51 +66,51 @@ public void Authenticate(Uri authorizationUri, Uri redirectUri, RequestContext r
-                                ContinueAuthentication(callbackUrl.ToString());
-                            }
-                        });
+                //if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
```
> //if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0)) [](start = 15, length = 56)

argh.... doesn't seem good. #Closed

trwalke replied:
These are left over changes from testing. will be reverted #Closed

henrik-me commented on core/src/Platforms/iOS/WebviewBase.cs:
```diff
@@ -40,7 +40,7 @@ internal abstract class WebviewBase : NSObject, IWebUI, ISFSafariViewControllerD
         protected SFSafariViewController safariViewController;
         protected SFAuthenticationSession sfAuthenticationSession;
-        protected ASWebAuthenticationSession asWebAuthenticationSession;
+        //protected ASWebAuthenticationSession asWebAuthenticationSession;
```
> //protected ASWebAuthenticationSession asWebAuthenticationSession; [](start = 7, length = 67)

not good #Closed

trwalke replied:
this should not have been checked in. reverting #Closed

henrik-me commented on core/src/Microsoft.Identity.Core.csproj:
```diff
@@ -1,8 +1,15 @@
+    <Message Text="MAC_OS_BUILD = $(MAC_OS_BUILD)" Importance="High" />
+  </Target>
+  <PropertyGroup Condition="'$(MAC_OS_BUILD)' == ''">
+    <TargetFrameworks>net45;netstandard1.1;netstandard1.3;uap10.0;MonoAndroid8.1;Xamarin.iOS10;netcoreapp1.0</TargetFrameworks>
```
>netcoreapp1.0< [](start = 95, length = 14)

mac should/could also include netcore ... possibly also android #Pending

trwalke replied:
I will try compiling with these two targets #Pending

henrik-me replied:
nit:
you should be able to keep android as well. reason for this ask is to ensure that we can use the solution for as much as possible on a Mac (should someone prefer that as their dev setup)... as it's not needed for the test setup you could also enable just what is needed for the iOS setup....

---
In reply to: [235214338](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1418#discussion_r235214338) [](ancestors = 235214338)

henrik-me commented on core/tests/Test.Microsoft.Identity.Core.UIAutomation/MSALMobileTestHelper.cs:
```diff
@@ -219,5 +219,26 @@ public void PerformB2CLocalAccountSignInFlow(ITestController controller, LabUser
+            AcquireTokenInteractiveTestHelper(
+                controller,
+                labResponse,
+                CoreUiTestConstants.UIBehaviorConsent);
```
>UIBehaviorConsent [](start = 36, length = 17)

IMO, not obvious this is the default, consider adding that to the method name #Pending

henrik-me commented on msal/src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:
```diff
@@ -1,7 +1,14 @@
     <TargetFrameworks>net45;netstandard1.3;uap10.0;MonoAndroid8.1;Xamarin.iOS10;netcoreapp1.0</TargetFrameworks>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(MAC_OS_BUILD)' != ''">
+    <TargetFrameworks>netstandard1.3;Xamarin.iOS10</TargetFrameworks>
```
>netstandard1.3;Xamarin.iOS10 [](start = 22, length = 28)

see general comment on which targets to include for mac #Pending

henrik-me replied:
see comment elsewhere

---
In reply to: [235163880](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1418#discussion_r235163880) [](ancestors = 235163880)

henrik-me reviewed (changes requested):
:clock1:

bgavrilMS commented on adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj:
```diff
@@ -1,7 +1,14 @@
+  <Target Name="MacOSBuild" BeforeTargets="Build">
+    <Message Text="MAC_OS_BUILD = $(MAC_OS_BUILD)" Importance="High" />
+  </Target>
+  <PropertyGroup Condition="'$(MAC_OS_BUILD)' == ''">
```
You have an msbuild variable that you can use - see https://github.com/Microsoft/msbuild/issues/539 #Closed

trwalke replied:
Great. ill use this instead #Closed

jennyf19 commented on core/src/Platforms/iOS/SystemWebview/SystemWebUI.cs:
```diff
@@ -66,51 +66,51 @@ public void Authenticate(Uri authorizationUri, Uri redirectUri, RequestContext r
-                                ContinueAuthentication(callbackUrl.ToString());
-                            }
-                        });
+                //if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
```
> //if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0)) [](start = 15, length = 56)

Why is this commented out? We need to support different iOS versions...ASWebAuthSession is for 12+, SFAuthSession is for 11 (now deprecated) and SFSafariViewController is for everything else. this is for system browser, and to be on-par with iOS team, so it can't be commented out. let's discuss other options. #Closed

trwalke replied:
These are left over changes from testing. will be reverted #Closed

jennyf19 commented on msal/tests/Test.MSAL.NET.UIAutomation/XamariniOSADALTests.cs:
```diff
@@ -0,0 +1,132 @@
+    /// Configures environment for core/android tests to run
+    /// </summary>
+    [TestFixture(Platform.iOS)]
+    public class XamarinDroidADALTests
```
>XamarinDroidADALTests [](start = 17, length = 21)

should this be iOS? #Closed

jennyf19 commented on msal/tests/Test.MSAL.NET.UIAutomation/XamariniOSMSALTests.cs:
```diff
@@ -0,0 +1,166 @@
+    /// Configures environment for core/android tests to run
+    /// </summary>
+    [TestFixture(Platform.iOS)]
+    public class XamarinMSALDroidTests
```
>Droid [](start = 28, length = 5)

iOS? #Closed

jennyf19 commented on msal/tests/Test.MSAL.NET.UIAutomation/XamariniOSMSALTests.cs:
```diff
@@ -0,0 +1,166 @@
+    /// Configures environment for core/android tests to run
+    /// </summary>
+    [TestFixture(Platform.iOS)]
+    public class XamarinMSALDroidTests
```
> XamarinMSALDroidTests [](start = 16, length = 22)

Not a fan of this class name: `XamariniOSMSALTests.cs` it's hard to read. Same w/the ADAL one.
some ideas:
`XamarinIOSMsalTests.cs`
`IOSXamarinMsalTests.cs`
`iOSXamarinMsalTests.cs` -> we already don't following naming convention, so i guess this is okay

jennyf19 replied:
Can we do XamarinIOSMsalTests? Also, the name of the file needs to be changed...or it's not showing up as changed...same w/respective ADAL class.

---
In reply to: [235181658](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1418#discussion_r235181658) [](ancestors = 235181658)

henrik-me replied:
@trwalke pls. consider these suggestions.

---
In reply to: [235236782](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1418#discussion_r235236782) [](ancestors = 235236782,235181658)

jennyf19 reviewed (changes requested):
few questions. will be great to get iOS automation going.

Commit:
Refactoring names of classes/methods.
adding android and net core targets to macOS build.
added MSBuild provided property to detect OS.

jennyf19 commented on adal/devApps/XFormsApp.iOS/AppDelegate.cs:
```diff
@@ -52,6 +52,9 @@ public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsAppli
         {
             global::Xamarin.Forms.Forms.Init();
+            #if ENABLE_TEST_CLOUD
+            Xamarin.Calabash.Start();
```
>Calabash [](start = 20, length = 8)

I read on app center tonight that they are deprecating Calabash [https://docs.microsoft.com/en-us/appcenter/test-cloud/preparing-for-upload/calabash](https://docs.microsoft.com/en-us/appcenter/test-cloud/preparing-for-upload/calabash) ...should we use something else? #Closed

jennyf19 replied:
nevermind...i see they are still supporting it through Xamarin.UITest, which we are using. seems weird. lol.

---
In reply to: [235236921](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1418#discussion_r235236921) [](ancestors = 235236921)

trwalke replied:
Yeah. I saw the same. scared me for a second. lol
 #Closed

jennyf19 commented on msal/tests/Test.MSAL.NET.UIAutomation/XamariniOSADALTests.cs:
```diff
@@ -0,0 +1,132 @@
+    /// Configures environment for core/android tests to run
+    /// </summary>
+    [TestFixture(Platform.iOS)]
+    public class XamarinIOSADALTests
```
>XamarinIOSADALTests [](start = 17, length = 19)

how about `iOSAdalTests` or `XamarinIOSAdalTests` we probably don't need Xamarin named explicitly.

Commit:
Removing mono droid from macOS build

Commit:
Removing netcore target from macOS build

Commit:
Reenabling Netcore on macOS build.
Fixing targeting issue.

Commit:
adding target for macOS build

henrik-me commented on adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj:
```diff
@@ -1,7 +1,14 @@
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(OS)' == 'OSX'">
+    <TargetFrameworks>netstandard1.1;netstandard1.3;Xamarin.iOS10;netcoreapp1.0</TargetFrameworks>
+  </PropertyGroup>
```
pls. align in all the relevant projects (there are minor differences right now)

henrik-me commented on adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidADALTests.cs:
```diff
@@ -41,7 +41,7 @@ namespace Test.ADAL.UIAutomation
     /// Configures environment for core/android tests to run
     /// </summary>
-    [TestFixture(Platform.Android)]
+    [TestFixture(Platform.iOS)]
```
>iOS [](start = 26, length = 3)

why iOS? ... is the file name wrong? in that case the class name is also wrong. #Closed

trwalke replied:
Wrong platform in file #Closed

henrik-me commented on adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidMSALTests.cs:
```diff
@@ -148,7 +131,7 @@ public void B2CLocalAccountAcquireTokenSilentTest()
         public void AcquireTokenADFSV4InteractiveFederatedTest()
         {
-            _msalMobileTestHelper.AcquireTokenInteractiveTestHelper(
+            _msalMobileTestHelper.AcquireTokenInteractiveTestHelperWithLoginBehaviorDefault(
```
>AcquireTokenInteractiveTestHelperWithLoginBehaviorDefault [](start = 34, length = 57)

I intended this not to say default but instead with Consent... as that is what is being used as default? #Closed

henrik-me commented on msal/tests/Test.MSAL.NET.UIAutomation/XamariniOSADALTests.cs:
```diff
@@ -0,0 +1,132 @@
+        ITestController xamarinController = new XamarinUITestController();
+        ADALMobileTestHelper _adalMobileTestHelper = new ADALMobileTestHelper();
+
+        public XamarinIOSADALTests(Platform platform)
```
> public XamarinIOSADALTests(Platform platform) [](start = 7, length = 46)

any reason the ctr. doesn't have reference docs... everthing else seems to be good.

Commit:
Refactoring file names/method names

Commit:
Merge remote-tracking branch 'origin/dev' into trwalke/iOSAppCenterTest

# Conflicts:
#	adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidMSALTests.cs
#	msal/src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj

Commit:
Updating iOS tests after merge

Commit:
Refactoring method

trwalke commented:
msal/src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj
msal/tests/Test.MSAL.NET.UIAutomation/MSALMobileTestHelper.cs
msal/tests/Test.MSAL.NET.UIAutomation/XamarinDroidMSALTests.cs

Commit:
Configuring Mac Build

Commit:
Configuring MacOS Build

Commit:
Test

Commit:
Adding msbuild property back

Commit:
removing targets from mac

Commit:
Adding build targets

Commit:
fixing error

Commit:
Upgrading xamarin

henrik-me commented:
Generally looks ok. Would like to see a test run which passes all test cases.

Commit:
Disabling iOS 12 code on macOS build

Commit:
Resolving return type errors on xamarin build

Commit:
Merge remote-tracking branch 'origin/trwalkeBuildtest' into trwalke/iOSAppCenterTest

Commit:
Fixing build errors

Commit:
Fixing build error

Commit:
Downgrading Xamarin to resolve build issue

Commit:
Revert "Downgrading Xamarin to resolve build issue"

This reverts commit 304d056b3c3d666f318946d1a495d9411834651c.

Commit:
Revert "Fixing build error"

This reverts commit 8a2b2b01804bbd28d0e5462e6358f3e87549cf8a.

MarkZuber commented on adal/devApps/XFormsApp/App.cs:
```diff
@@ -52,7 +52,7 @@ protected override void OnStart()
             AppCenter.Start("android=3ea693de-dcb4-4b65-a1b5-04ad2f71ada0;" +
                   "uwp={Your UWP App secret here};" +
-                  "ios={Your iOS App secret here}",
+                  "ios={4be201cc-2027-4045-b535-d57de5ff36c7}",
```
is this a secret?  should this be checked in?  Since it's in a PR if it is a secret, we'll need to roll it...

trwalke replied:
not necessarily a secret, more of an identifier for our app in appcenter. The actual credentials required to access appcenter is stored securely in our Azure DevOps

Commit:
Reverting Xamarin Forms Upgrade

Commit:
Splitting adal/msal mobile test projects
